Update to Mockito 4.5.1

In the edelta.maven.plugin we get this error:

```
Error:  Tests run: 9, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 17.019 s <<< FAILURE! - in edelta.maven.plugin.EdeltaMojoTest
Error:  testDebugLogging(edelta.maven.plugin.EdeltaMojoTest)  Time elapsed: 4.173 s  <<< ERROR!
java.lang.IllegalStateException: Could not initialize plugin: interface org.mockito.plugins.MockMaker (alternate: null)
	at edelta.maven.plugin.EdeltaMojoTest.spyLog(EdeltaMojoTest.java:189)
	at edelta.maven.plugin.EdeltaMojoTest.testDebugLogging(EdeltaMojoTest.java:162)
Caused by: java.lang.IllegalStateException: Internal problem occurred, please report it. Mockito is unable to load the default implementation of class that is a part of Mockito distribution. Failed to load interface org.mockito.plugins.MockMaker
	at edelta.maven.plugin.EdeltaMojoTest.spyLog(EdeltaMojoTest.java:189)
	at edelta.maven.plugin.EdeltaMojoTest.testDebugLogging(EdeltaMojoTest.java:162)
Caused by: java.lang.reflect.InvocationTargetException
	at edelta.maven.plugin.EdeltaMojoTest.spyLog(EdeltaMojoTest.java:189)
	at edelta.maven.plugin.EdeltaMojoTest.testDebugLogging(EdeltaMojoTest.java:162)
Caused by: java.lang.ExceptionInInitializerError
	at edelta.maven.plugin.EdeltaMojoTest.spyLog(EdeltaMojoTest.java:189)
	at edelta.maven.plugin.EdeltaMojoTest.testDebugLogging(EdeltaMojoTest.java:162)
Caused by: java.lang.SecurityException: class "javax.annotation.Nonnull"'s signer information does not match signer information of other classes in the same package
	at edelta.maven.plugin.EdeltaMojoTest.spyLog(EdeltaMojoTest.java:189)
	at edelta.maven.plugin.EdeltaMojoTest.testDebugLogging(EdeltaMojoTest.java:162)

Error:  testProjectWithErrorFailOnValidationErrorFalse(edelta.maven.plugin.EdeltaMojoTest)  Time elapsed: 0.11 s  <<< ERROR!
java.lang.IllegalStateException: Could not initialize plugin: interface org.mockito.plugins.MockMaker (alternate: null)
	at edelta.maven.plugin.EdeltaMojoTest.spyLog(EdeltaMojoTest.java:189)
	at edelta.maven.plugin.EdeltaMojoTest.testProjectWithErrorFailOnValidationErrorFalse(EdeltaMojoTest.java:146)
Caused by: java.lang.IllegalStateException: Internal problem occurred, please report it. Mockito is unable to load the default implementation of class that is a part of Mockito distribution. Failed to load interface org.mockito.plugins.MockMaker
	at edelta.maven.plugin.EdeltaMojoTest.spyLog(EdeltaMojoTest.java:189)
	at edelta.maven.plugin.EdeltaMojoTest.testDebugLogging(EdeltaMojoTest.java:162)
Caused by: java.lang.reflect.InvocationTargetException
	at edelta.maven.plugin.EdeltaMojoTest.spyLog(EdeltaMojoTest.java:189)
	at edelta.maven.plugin.EdeltaMojoTest.testDebugLogging(EdeltaMojoTest.java:162)
Caused by: java.lang.ExceptionInInitializerError
	at edelta.maven.plugin.EdeltaMojoTest.spyLog(EdeltaMojoTest.java:189)
	at edelta.maven.plugin.EdeltaMojoTest.testDebugLogging(EdeltaMojoTest.java:162)
Caused by: java.lang.SecurityException: class "javax.annotation.Nonnull"'s signer information does not match signer information of other classes in the same package
	at edelta.maven.plugin.EdeltaMojoTest.spyLog(EdeltaMojoTest.java:189)
	at edelta.maven.plugin.EdeltaMojoTest.testDebugLogging(EdeltaMojoTest.java:162)
```

--- a/.mvn/wrapper/maven-wrapper.properties
+++ b/.mvn/wrapper/maven-wrapper.properties
@@ -1,2 +1,2 @@
-distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.8.4/apache-maven-3.8.4-bin.zip
+distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.8.5/apache-maven-3.8.5-bin.zip
 wrapperUrl=https://repo.maven.apache.org/maven2/io/takari/maven-wrapper/0.5.6/maven-wrapper-0.5.6.jar

--- a/edelta.parent/edelta.maven.plugin/pom.xml
+++ b/edelta.parent/edelta.maven.plugin/pom.xml
@@ -31,6 +31,19 @@
       <groupId>io.github.lorenzobettini.edelta</groupId>
       <artifactId>edelta.lib</artifactId>
       <exclusions>
+        <!-- 
+          https://github.com/LorenzoBettini/edelta/issues/429
+
+          Exclude it to avoid java.lang.SecurityException when running in a Tycho build:
+          java.lang.SecurityException:
+          class "javax.annotation.Nonnull"'s signer information does not match
+          signer information of other classes in the same package
+          at edelta.maven.plugin.EdeltaMojoTest.spyLog(EdeltaMojoTest.java)
+
+          This dep is required by guava 30.1-jre
+            required by org.eclipse.xtext.xbase.lib
+              required by edelta.lib
+        -->
         <exclusion>
           <groupId>com.google.code.findbugs</groupId>
           <artifactId>jsr305</artifactId>

--- a/edelta.parent/edelta.maven.plugin/pom.xml
+++ b/edelta.parent/edelta.maven.plugin/pom.xml
@@ -11,9 +11,9 @@
   <packaging>maven-plugin</packaging>
 
   <properties>
-    <maven.version>3.8.1</maven.version>
+    <maven.version>3.8.3</maven.version>
     <assertj-version>3.22.0</assertj-version>
-    <mockito-version>3.12.4</mockito-version>
+    <mockito-version>4.5.1</mockito-version>
   </properties>
 
   <prerequisites>
@@ -30,6 +30,12 @@
         since maven-invoker-plugin does not install this locally -->
       <groupId>io.github.lorenzobettini.edelta</groupId>
       <artifactId>edelta.lib</artifactId>
+      <exclusions>
+        <exclusion>
+          <groupId>com.google.code.findbugs</groupId>
+          <artifactId>jsr305</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
     <dependency>
       <groupId>org.apache.maven</groupId>

--- a/edelta.parent/edelta.target/edelta.target.target
+++ b/edelta.parent/edelta.target/edelta.target.target
@@ -34,12 +34,12 @@
 <location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
 <repository location="https://download.eclipse.org/tools/orbit/downloads/2022-03/"/>
 </location>
-	<location includeDependencyDepth="infinite" includeDependencyScopes="compile" includeSource="true" missingManifest="generate" type="Maven">
+	<location includeDependencyDepth="infinite" includeDependencyScopes="compile,runtime" includeSource="true" missingManifest="generate" type="Maven">
 		<dependencies>
 			<dependency>
 				<groupId>org.mockito</groupId>
 				<artifactId>mockito-core</artifactId>
-				<version>3.12.4</version>
+				<version>4.5.1</version>
 				<type>jar</type>
 			</dependency>
 		</dependencies>

